hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/states/datanode/InitDatanodeState.java

```diff
@@ -125,7 +125,7 @@ private void persistContainerDatanodeDetails() {
     File idPath = new File(dataNodeIDPath);
     DatanodeDetails datanodeDetails = this.context.getParent()
         .getDatanodeDetails();
-    if (datanodeDetails != null && !idPath.exists()) {
```

Comment:
What's the motivation for dropping this check?

Reply:
This is because when the datanode got restarted in k8s, the IP will be changed. So the original info in this file is not accurate any more. This will make sure we update with the latest info. 

And when we are not using k8s, I think it is not harmful to always update this file whenever the node restarts.